--- configure	2005-06-17 17:38:49.073838400 +1000
+++ configure.new	2005-06-17 17:46:08.434683200 +1000
@@ -10566,6 +10566,13 @@ else
   want_bzip2="yes"
 fi;
 
+case $host_os in
+ cygwin*)
+ # add -lgdi32 -lwinmm for cygwin 
+ # (maybe we need to include  | mingw* | pw32* as well
+ 	LIBCLAMAV_LIBS="$LIBCLAMAV_LIBS -lgdi32 -lwinmm";
+esac
+ 
 if test "$want_bzip2" = "yes"
 then
     echo "$as_me:$LINENO: checking for bzReadOpen in -lbz2" >&5
